### ISSUE TITLE
`search_data` results for `OPERA_L2_CSLC-S1_V1` are much larger than expected

### ISSUE THREAD

**Issue by scottstanie** [nsidc/earthaccess#746]
I was checking how hard it woudl be to use `earthaccess` to download some OPERA CSLC products (taking the small search from [here](https://github.com/nsidc/earthaccess/issues/439) as an example)

```python
import earthaccess
earthaccess.login()

results = earthaccess.search_data(
    short_name='OPERA_L2_CSLC-S1_V1',
    cloud_hosted=True,
    bounding_box=(-117.33, 35.541, -117.880, 35.991),
    temporal=("2023-10-01", "2023-10-15"),
    count=500
)
Granules found: 2737

```

This is a small bounding and only 15 days, so I'd expect just a few results.

Entering this into ASF (or using `asf_search`) [yields the expected 12 granules](https://search.asf.alaska.edu/#/?polygon=POLYGON((-117.88%2035.541,-117.33%2035.541,-117.33%2035.991,-117.88%2035.991,-117.88%2035.541))&start=2023-10-01T04:00:00Z&end=2023-10-15T03:59:59Z&dataset=OPERA-S1&productTypes=CSLC&resultsLoaded=true&granule=OPERA_L2_CSLC-S1_T071-151217-IW3_20231011T135232Z_20240505T012143Z_S1A_VV_v1.1&zoom=7.946&center=-116.535,34.551)

I'm not sure if this is something odd or unexpected with the OPERA data compared to others, or how earthaccess would match this many granules.

**Comment by scottstanie** [comment]
Shoot I'm sorry, I just copied those parameters to test, but I didn't notice that the bounding box from that issue was incorrect and had the order (right, bottom, left, top).

Using `bounding_box=(-117.33, 35.541, -117.880, 35.991)` returns 10 results. 

I actually saw this because the `asf_search` library printed the helpful warning 
```
WARNING:asf_search:WKT REPAIR/VALIDATION: The following repairs were performed on the provided AOI:
["'type': 'REVERSE': 'report': Reversed polygon winding order"]
```
Not sure if that's worth adding here or not, but it helped me pick out that problem.
